caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -0,0 +1,384 @@
+            documents=documents,
+            top_n=top_n,
+            return_documents=return_documents,
+            return_queries=return_query,
+            return_text=return_text,
+        ).results
+
+        return (
+            results[0]
+            if len(results) > 0
+            else RerankQueryResult(scores=[], query=query if return_query else None)
+        )
+
+    @RerankTasks.taskmethod()
+    def run_rerank_queries(
+        self,
+        queries: List[str],
+        documents: List[JsonDict],
+        top_n: Optional[int] = None,
+        return_documents: bool = True,
+        return_queries: bool = True,
+        return_text: bool = True,
+    ) -> RerankPredictions:
+        """Rerank the documents returning the most relevant top_n in order for each of the queries.
+        Args:
+            queries: List[str]
+                Each of the queries will be compared to the text of each of the documents.
+            documents:  List[JsonDict]
+                Each document is a dict. The text value is used for comparison to the query.
+                If there is no text key, then _text is used and finally default is "".
+            top_n:  Optional[int]
+                Results for the top n most relevant documents will be returned.
+                If top_n is not provided or (not > 0), then all are returned.
+            return_documents:  bool
+                Default True
+                Setting to False will disable returning of the input document (index is returned).
+            return_queries:  bool
+                Default True
+                Setting to False will disable returning of the query (results are in query order)
+            return_text:  bool
+                Default True
+                Setting to False will disable returning of document text string that was used.
+        Returns:
+            RerankPredictions
+                For each query in queries (in the original order)...
+                Returns the (top_n) scores in relevance order (most relevant first).
+                The results always include a score and index which may be used to find the document
+                in the original documents list. Optionally, the results also contain the entire
+                document with its score (for use in chaining) and for convenience the query and
+                text used for comparison may be returned.
+        """
+
+        error.type_check(
+            "<NLP09038249E>",
+            list,
+            queries=queries,
+            documents=documents,
+        )
+
+        if len(queries) < 1 or len(documents) < 1:
+            return RerankPredictions([])
+
+        if top_n is None or top_n < 1:
+            top_n = len(documents)
+
+        # Using input document dicts so get "text" else "_text" else default to ""
+        def get_text(srd):
+            return srd.get("text") or srd.get("_text", "")
+
+        doc_texts = [get_text(srd) for srd in documents]
```

Comment:
Took part of your advice (srd -> doc) but I kept get_text function because that decision to use text else _text else "" is in 2 places and I do think we will reconsider that default part.  So at least it is only in once place.